lib/classification_dump_cache.rb

```diff
@@ -49,6 +56,13 @@ def secure_user_ip(ip_string)
   def find_workflow_at_version(workflow, major_version, minor_version)
     workflow.workflow_versions.where("major_number >= ? AND minor_number >= ?", major_version, minor_version).order("major_number ASC, minor_number ASC").first!
   rescue ActiveRecord::RecordNotFound
-    workflow.workflow_versions.order("major_number ASC, minor_number ASC").last
+    fallback = workflow.workflow_versions.order("major_number ASC, minor_number ASC").last
+    return fallback if fallback.present?
+
+    if workflow.major_version == major_version && workflow.minor_version == minor_version
```

Comment:
What's the difference between this and the `.where` above? Is it just to confirm that it doesn't exist to allow you to raise, or is there some circumstance I'm not thinking about where this would be able to build after no version (or fallback) was found?

Reply:
The where finds workflow_version records. If that doesn't exist, then this checks if the requested version perhaps happens to be the latest one (the workflow itself has a major/minor number equal to requested). In that case we're lucky and can make a workflow_version from the current workflow state.